src/generic.jl

```diff
@@ -892,6 +892,89 @@ opnorm(v::TransposeAbsVec) = norm(v.parent)
 
 norm(v::AdjOrTrans, p::Real) = norm(v.parent, p)
 
+"""
+    lpdist(x::T, y::T, p::Int) where {T <: AbstractVector}
+    lpdist(x::T, y::T, p::Real) where {T <: AbstractVector}
+
+    Computes the Lₚ distance between two vectors `x` and `y`, defined as:
+    for 0 < p < Inf: Lₚ dist = ᵖ√(Σ|xᵢ - yᵢ|ᵖ)
+    for p = 0: count of mismatches for p = 0
+    for p + Inf: L_∞ = maxᵢ(abs(xᵢ - yᵢ))
+"""
+
+function _lpunnorm(x::T, y::T, fn) where {T}
+    r = 0.0
+    for i in eachindex(x,y)
+        @inbounds r += fn(x[i] - y[i])
+    end
+
+    return r
+end
+
+
+function lpdist(x::T, y::T, p::Int) where {T <: AbstractVector}
+    p < 0 && throw(DomainError("p must be non-negative"))
+    length(x) == length(y) || throw(DimensionMismatch("x and y have different lenghts"))
+    if p == 0
+        @warn("Technically not a distance metric for p = 0")
+        fn = !iszero # simply count number of nonzeros
+    elseif p == 1
+        fn = abs
+    elseif p == 2
+        fn = abs2
```

Comment:
Should be `norm` and `norm_sqr`, respectively.

Reply:
Is `norm` better than `abs` for a scalar too?